backend/src/govsg/services/govsg-callback.service.ts

```diff
@@ -142,12 +145,24 @@ const storePrecreatedPasscode = async (
 const parseTemplateMessageWebhook = async (
   body: WhatsAppTemplateMessageWebhook,
   clientId: WhatsAppApiClient
+): Promise<void> => {
+  await Promise.all(
```

Comment:
I think these are sequential events so if they come in bulk we should process them sequentially in the order that they come in. Or better yet, if there're event timestamps available we should follow those

Reply:
I agree with you, Stanley. On it.

Reply:
eh if I'm not wrong `body` always return a `statuses` array of length 1 (but no issue with handling it this way)

Reply:
@tan-zx are you able to point me to the docs or comms from Meta that indicate this?

Why is their design so ambiguous!!!!

They have [a small section in their docs](https://developers.facebook.com/docs/graph-api/webhooks/getting-started#validate-payloads) which say batching may occur.

Reply:
@stanleynguyen and yup done with sequential processing after sorting by timestamp.

Reply:
> They have [a small section in their docs](https://developers.facebook.com/docs/graph-api/webhooks/getting-started#validate-payloads) which says batching may occur.

the link is definitely not for On Prem API (look at how they describe the configuration of webhooks)
![image](https://github.com/opengovsg/postmangovsg/assets/67887489/fcbb7599-bfb5-4edd-9d28-54bee2f390dc)


> are you able to point me to the docs or comms from Meta that indicate this i.e. always array of length 1?
> Why is their design so ambiguous!!!!

it's documented somewhere on their site (I took 2 mins to try to find but couldn't, but I rmb it's there somewhere).

when I worked on it, I've documented [what their docs said vs what I actually received here](https://www.notion.so/opengov/WhatsApp-On-Prem-API-Webhook-f7bb4341658048ab8e525b0c32d3ea1a?pvs=4#f70cca05e7d74b4f847a6e1d72ea5617). as you can see, there are actually discrepancies between their docs and what their API does. I've given this feedback to the Meta team (you can find this in the support chat history).

Reply:
Thank you @tan-zx for sharing! I'll close this PR for now.